test/LaunchDarkly.EventSource.Tests/Stubs/StubHandler.cs

```diff
@@ -26,41 +28,154 @@ public class StubMessageHandler : HttpMessageHandler
-            }
-            
-            return Task.FromResult(response);
+            return Task.FromResult(response.MakeResponse(cancellationToken));
         }
 
+        public void QueueResponse(StubResponse response) => _responses.Enqueue(response);
+        
+        public IEnumerable<HttpRequestMessage> GetRequests() =>
+            _requests;
+    }
+
+    public abstract class StubResponse
+    {
+        public abstract HttpResponseMessage MakeResponse(CancellationToken cancellationToken);
+
+        protected StubResponse() { }
+
+        public static StubResponse WithResponse(HttpResponseMessage message)
+        {
+            return new StubResponseWithHttpResponse(message);
+        }
+
+        public static StubResponse WithStatus(HttpStatusCode status)
+        {
+            return new StubResponseWithHttpResponse(new HttpResponseMessage(status));
+        }
+
+        public static StubResponse WithIOError()
+        {
+            return new StubResponseWithIOError();
+        }
+
+        public static StubResponse StartStream(params StreamAction[] actions)
+        {
+            return new StubResponseWithStream(actions);
+        }
+    }
 
-        public void QueueResponse(HttpResponseMessage response) =>
-            _responses.Enqueue(response);
+    internal class StubResponseWithHttpResponse : StubResponse
+    {
+        private readonly HttpResponseMessage message;
+
+        public StubResponseWithHttpResponse(HttpResponseMessage message)
+        {
+            this.message = message;
+        }
+
+        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
+        {
+            return message;
+        }
+    }
 
-        public void QueueStringResponse(string content)
-        {
-            var response =
-                new HttpResponseMessage(HttpStatusCode.OK)
-                {
-                    Content = new StringContent(content, System.Text.Encoding.UTF8, "text/event-stream")
-                };
+    internal class StubResponseWithIOError : StubResponse
+    {
+        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
+        {
+            throw new HttpRequestException("Unit Test Exception Message");
+        }
+    }
 
-            _responses.Enqueue(response);
+    internal class StubResponseWithStream : StubResponse
+    {
+        private readonly StreamAction[] _actions;
+
+        public StubResponseWithStream(StreamAction[] actions)
+        {
+            _actions = actions;
+        }
+
+        override public HttpResponseMessage MakeResponse(CancellationToken cancellationToken)
+        {
+            var httpResponse = new HttpResponseMessage(HttpStatusCode.OK);
+            var streamRead = new AnonymousPipeServerStream(PipeDirection.In);
+            var streamWrite = new AnonymousPipeClientStream(PipeDirection.Out, streamRead.ClientSafePipeHandle);

```

Comment:
This is the part that required me to go up to .NET Core 2.0. It's possible that there is some other way to make a blocking stream that reads from another stream.